emodelrunner/protocols/reader.py

```diff
@@ -0,0 +1,691 @@
+            "RinHoldcurrent_hyp",
+            rin_protocol_template=self.protocols_dict["Rin_hyp"],
+            holdi_estimate_multiplier=protocol_definitions["RinHoldcurrent_hyp"][
+                "holdi_estimate_multiplier"
+            ],
+            holdi_precision=protocol_definitions["RinHoldcurrent_hyp"][
+                "holdi_precision"
+            ],
+            holdi_max_depth=protocol_definitions["RinHoldcurrent_hyp"][
+                "holdi_max_depth"
+            ],
+            prefix=prefix,
+        )
+
+        other_protocols = []
+
+        for protocol_name in protocol_definitions["Main"]["other_protocols"]:
+            other_protocols.append(self.protocols_dict[protocol_name])
+
+        pre_protocols = []
+        preprot_score_threshold = 1
+
+        if "pre_protocols" in protocol_definitions["Main"]:
+            for protocol_name in protocol_definitions["Main"]["pre_protocols"]:
+                pre_protocols.append(self.protocols_dict[protocol_name])
+            preprot_score_threshold = protocol_definitions["Main"][
+                "preprot_score_threshold"
+            ]
+
+        runopt = False
+        self.protocols_dict["Main"] = thalamus_protocols.RatSSCxMainProtocol(
+            "Main",
+            rmp_protocol=self.protocols_dict["RMP"],
+            rmp_score_threshold=protocol_definitions["Main"]["rmp_score_threshold"],
+            rinhold_protocol_dep=rinhold_protocol_dep,
+            rinhold_protocol_hyp=self.protocols_dict["RinHoldcurrent_hyp"],
+            rin_score_threshold=protocol_definitions["Main"]["rin_score_threshold"],
+            thdetect_protocol_dep=thdetect_protocol_dep,
+            thdetect_protocol_hyp=self.protocols_dict["ThresholdDetection_hyp"],
+            other_protocols=other_protocols,
+            pre_protocols=pre_protocols,
+            preprot_score_threshold=preprot_score_threshold,
+            use_rmp_rin_thresholds=runopt,
+        )
+
+    def parse_sscx_protocols(
+        self,
+        protocols_filepath,
+        stochkv_det=None,
+        prefix="",
+        apical_point_isec=-1,
+        syn_locs=None,
+    ):
+        """Parses the SSCX protocols from the json file input.
+
+        Args:
+            protocols_filename (str): path to the protocols file
+            stochkv_det (bool): set if stochastic or deterministic
+            prefix (str): prefix used in naming responses, features, recordings, etc.
+            apical_point_isec (int): apical point section index
+                Should be given if there is "somadistanceapic" in "type"
+                of at least one of the extra recordings
+            syn_locs (list of ephys.locations.NrnPointProcessLocation):
+                locations of the synapses (if any, else None)
+
+        Returns:
+            dict containing the protocols
+        """
+        protocol_definitions = self.load_protocol_json(protocols_filepath)
+
+        # fmt: off
```

Comment:
It's a way to tell `black` formatter not to format the following lines. When there's a long list, black makes it many lines. Here I used it for this list. `black` would make it ~10 lines otherwise :)

```
        protocols_with_type = [
            "StepProtocol", "StepThresholdProtocol", "RampThresholdProtocol",
            "RampProtocol", "RatSSCxThresholdDetectionProtocol", "Vecstim", "Netstim"]

```
Here's a good example of when to use it
https://stackoverflow.com/a/58584557/1935611